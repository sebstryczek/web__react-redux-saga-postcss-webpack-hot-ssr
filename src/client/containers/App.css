:root {
  --body-background: purple;
  --child-background: gold;
}
body {
  background: var(--body-background);
  & * {
    background: var(--child-background);
  }
}