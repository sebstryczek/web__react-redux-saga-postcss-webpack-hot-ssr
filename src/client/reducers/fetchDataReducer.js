import * as actions from '../constants/actionTypes';

const reducer = (state = {}, action) => {
  switch (action.type) {

    case actions.FETCH_DATA_REQUEST:
      return state;

    case actions.FETCH_DATA_SUCCESS:
      return {...state, data: action.payload};

    default:
      return state;
      
  }
}

export default reducer;
