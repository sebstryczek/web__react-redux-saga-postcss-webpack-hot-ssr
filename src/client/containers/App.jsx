import React from 'react';
import { Switch, BrowserRouter, Route } from 'react-router-dom';
import { renderRoutes } from 'react-router-config';
import { connect } from 'react-redux';

import routes from '../routes';
import Page3 from './Page3';
import Example from '../components/Example';
import ExampleStyledComponent from '../components/ExampleStyledComponent';

import { fetchData } from '../actions/fetchDataActions';

import './App.css';

class App extends React.Component {
  static fetchData(store) {
    return store.dispatch(fetchData());
  }

  componentDidMount() {
    this.props.fetchData();
  }

  componentWillReceiveProps(nextProps) {
    console.log(nextProps)
  }

  render() {
    const data = this.props.data || [];
    return (
      <div>
        <h1>App</h1>
        <Example />
        <ExampleStyledComponent />
        {
          this.props.data && this.props.data.map(
            (item, i) => <p key={i}>{item}</p>
          )
        }
        { renderRoutes(routes[0].routes) }
        <Route path="/page3/" component={Page3} />
        <Route path="/page4/" render={() => <h2>Page 4</h2>} />
        <Route exact path="/page4/sub" render={() => <h2>Page 4 Sub</h2>} />
      </div>
    );
  }
};

const mapStateToProps = (state) => {
  return { ...state };
};

const mapDispatchToProps = (dispatch) => {
  return {
    fetchData: () => dispatch(fetchData())
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(App);
