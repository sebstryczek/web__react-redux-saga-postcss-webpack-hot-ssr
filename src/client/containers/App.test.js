import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import {createStore, applyMiddleware} from 'redux';
import thunk from 'redux-thunk';

import reducers from '../reducers/fetchDataReducer'
import App from './App';

import renderer from 'react-test-renderer';
import configureMockStore from 'redux-mock-store'
import { configure as enzymeConfigure, mount, shallow } from 'enzyme';
import EnzymeAdapter from 'enzyme-adapter-react-16';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);
enzymeConfigure({ adapter: new EnzymeAdapter() });

test('[EXAMPLE TEST] Two plus two is four', () => {
  expect(2 + 2).toBe(4);
});

test('[EXAMPLE TEST] Home page rendered', () => {
  const store = mockStore({});
  const component = renderer.create(
    <Provider store={store}>
      <MemoryRouter initialEntries={[ '/' ]}>
        <App />
      </MemoryRouter>
    </Provider>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
  //tree = component.toJSON();
  //expect(tree).toMatchSnapshot();
});

test('[EXAMPLE TEST] Application render proper H1 header', () => {
  const store = mockStore({});
  const component = mount(
    <Provider store={store}>
      <MemoryRouter initialEntries={[ '/' ]}>
        <App />
      </MemoryRouter>
    </Provider>
  );
  const h1 = component.find('h1');
  console.log(h1.debug());
  expect(h1.text()).toEqual('App');
});
