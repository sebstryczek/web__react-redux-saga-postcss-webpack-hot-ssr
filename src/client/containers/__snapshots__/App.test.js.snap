// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[EXAMPLE TEST] Home page rendered 1`] = `
<div>
  <h1>
    App
  </h1>
  <button>
    Example
  </button>
  <div>
    <button
      className="sc-bdVaJa gzRQLa"
    >
      Normal Button
    </button>
    <button
      className="sc-bdVaJa dCTgtY"
    >
      Primary Button
    </button>
  </div>
  <div>
    <h2>
      Home
    </h2>
  </div>
</div>
`;
